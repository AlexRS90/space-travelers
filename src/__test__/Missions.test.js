import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Mission from '../components/Missions';

afterEach(() => {
  cleanup();
});

describe('render mission to the DOM', () => {
  test('render mission component to the screen', async () => {
    render(<Provider store={store}><Mission /></Provider>);
    const missionElement = await screen.findByTestId('missions');
    expect(missionElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Provider store={store}><Mission /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
