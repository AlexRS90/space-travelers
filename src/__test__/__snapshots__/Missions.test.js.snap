// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render mission to the DOM matches snapshot 1`] = `
<div
  data-testid="missions"
>
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
